### PR TITLE
Integrations invoices models only

### PR DESCRIPTION
**Pull request by ivannovosad**
## Roadmap Task

👉  https://getlago.canny.io/feature-requests/p/integration-with-netsuite

## Context

Currently Lago does not support accounting integrations

## Description

This PR adds integration resource models, factories and specs

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -287,6 +288,18 @@ def should_assign_sequential_id?
     status_changed_to_finalized?
   end
 
+  def trigger_invoice_sync
```
Comment (vincent-pochet) [inline app/models/invoice.rb:291]: Not sure we should trigger business logic from the model. What do you think of having this called from the services?

Comment (ivannovosad) [reply app/models/invoice.rb:291]: Here in the [followup PR](https://github.com/getlago/lago-api/pull/2013/files) you can see that those methods should be called after save, so in this case I think it belongs in the model.

Comment (vincent-pochet) [reply app/models/invoice.rb:291]: That's my point :). Having callbacks in model to sync with external services is usually not a good practice.

I would suggest to add the logic in the service dealing with the create/update of the invoice, or in last resorts into an observer

Comment (vincent-pochet) [reply app/models/invoice.rb:291]: Happy to discuss it if you want :)

Comment (ivannovosad) [reply app/models/invoice.rb]: I changed it so the model checks if it's possible to sync, then all the services that handle saving invoices will need to call `should_sync_invoice?` and `should_sync_sales_order?` and enquee the jobs.

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**